invenio_rdm_migrator/state.py

```diff
@@ -281,11 +278,22 @@ def _initialize_db(cls, metadata):
 class StateEntity:
     """Helper class to query specific state tables."""
 
-    def __init__(self, state, table_name, pk_attr):
+    def __init__(self, state, table_name, pk_attr, cache=True):
         """Constructor."""
         self.state = state
         self.pk_attr = pk_attr
         self.table_name = table_name
+        self._cache = None
+        if cache:
+            self._cache = {}
+            self._init_cache()
+
+    def _init_cache(self):
+        """Initialize the cache from state."""
+        for row in self.state.all(self.table_name):
+            data = self._row_as_dict(row)
+            key = data[self.pk_attr]
```

Comment:
Correct, that looks wrong indeed... I can remove the PK from `value`, since anyways the lookups only happen using `key`.

Reply:
Actually... `value`'s columns type is `sa.Integer`? What happens when we store the secret keys there? :S

I was thinking maybe I should change it to string... but then I know that we also store the autoincrement integer PKs there... Maybe the solution is to "specialize" a table for PKs as in `STATE.PKS` and `STATE.VALUES`? In any case probably something to shelve.

Reply:
I honestly don't know, I have the feeling there was some rebase mistake at some point. But yes, the fact that its an integer I think is working by chance because the key is being stored as a number (but would be too long due to encoding)?

Reply:
Yeah, probably `bytes` but maybe SQLite has a trick under the hood